exercises/sam_jenkins/ex01_basics/src/mapFunctions.cpp

```diff
@@ -0,0 +1,48 @@
+#include <unordered_map>
+#include <sstream>
+#include <vector>
+#include "stringManip.h"
+
+
+
+using std::string; using std::unordered_map; using std::vector; using std::pair;
+
+unordered_map<string, int> countof(const string& input)
+{
+	//create a unordered map and string input stream
+	unordered_map<string, int>wordtable;
+	//create an iterator to loop through the string
+	auto i = input.begin();
```

Comment:
Two problems with the name `i`:
1) It's not descriptive.
2) Worse, typically it is used as the name for an integer counter variable in a c style for loop. 

`iter` would have been a better name (suggesting it's an iterator). An even better name would have described what were iterating over. `fileIterator`? `lineIterator`?